app-agent/src/exec_process.rs

```diff
@@ -33,6 +54,111 @@ pub fn exec_child(external_command: String, args: Vec<String>) -> anyhow::Result
     Ok(status)
 }
 
+fn handle_permission_denied(external_command: String) -> String {
+    let file_permission_denied = match File::open(external_command.clone()) {
+        Ok(file) => file,
+        Err(err) => {
+            panic!("Error when try to read the file: {}", err);
+        }
+    };
+    let metadata_file = file_permission_denied
+        .metadata()
+        .expect(format!("Unable to retrieve metadata for: {}", external_command).as_str());
+    // Check for user permissions.
+    let user_perm = match metadata_file.permissions().mode() & 0o500 {
+        0 => "rx",
+        1 => "r",
+        4 => "x",
+        _ => "",
+    };
+    // Check for group permissions.
+    let group_perm: &str = match metadata_file.permissions().mode() & 0o050 {
+        0 => "rx",
+        1 => "r",
+        4 => "x",
+        _ => "",
+    };
+    // Check for other permissions.
+    let other_perm = match metadata_file.permissions().mode() & 0o005 {
+        0 => "rx",
+        1 => "r",
+        4 => "x",
+        _ => "",
+    };
+    if user_perm == "rx" || group_perm == "rx" || other_perm == "rx" {
```

Comment:
What is all this doing, exactly? Isn't `chmod +rx` a solution that always work?

Reply:
I check which rights are missing, the goal is to suggest only to add missing rights, if read is missing but not execute, i will suggest `chmod +r` only

Reply:
That's very generous of your part, but `chmod` does the checking, no? If `chmod +r` works, `chmod +rx` will work too. Something that would really be helpful is, maybe, to check the permissions of the folder: what happens if the user wants to execute `/a/b/exec` but `/a/b` is not readable?

Reply:
Of course, if `+r` is missing, `+rx` will work too. I found it was more interesting for people to see that, for example, if only read right is missing, but the execute one is already present... But it's as you want, I can just suggest one chmod: `chmod +rx`.
About the parent folder, you're totally right, I will check this right in my next commit. I will only check the parent folder but not the parent of the parents, as it could be veryyyyyy long and not necessary, in my opinion